raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -314,8 +317,12 @@ def handle_secretrequest(
 
     already_received_secret_request = initiator_state.received_secret_request
 
+    # lock.amount includes the fees, transfer_description.amount is the actual
+    # payment amount, for the transfer to be valid and the unlock allowed the
+    # target must receive an amount between these values.
     is_valid_secretrequest = (
-        state_change.amount == initiator_state.transfer_description.amount and
+        state_change.amount <= lock.amount and
+        state_change.amount >= initiator_state.transfer_description.amount and
```

Comment:
We can't predict exactly what will be the schedule of each node in the path. The fees may get cheaper while the transfer is being sent, or the a shorter path exists that a mediator can take advantage. So I don't think this can be guaranteed to be equal.